File: BackEnd/todos-application/src/main/java/com/todos/service/TodosService.java
package com.todos.service;

import java.util.List;
import java.util.Map;

import com.todos.model.Todos;
import com.todos.result.ResultWrapper;

public interface TodosService {

	public ResultWrapper<Todos> save(Todos todos);

	public ResultWrapper<Boolean> delete(Integer id);

	public ResultWrapper<List<Todos>> getAll();

	public ResultWrapper<Todos> update(Todos todos);

	public ResultWrapper<Boolean> deleteMultiple(Map<String, List<Integer>> listOfIds);

}

File: FrontEnd/Todo-app-frontend/src/app/service/todos.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../environments/environment';



@Injectable({
  providedIn: 'root'
})
export class TodosService {

  constructor(private http: HttpClient) { }
  // Get All Todos
  getAllTodos() {
    return this.http.get(`${environment.baseUrl}getAllTodos`);
  }

  // To save the Todo Object
  saveTodos(todo: any) {
    return this.http.post(`${environment.baseUrl}saveTodos`, todo);
  }

  // Update Todo
  updateTodos(todo: any) {
    return this.http.put(`${environment.baseUrl}updateTodos`, todo);
  }

  // Delete Todo
  deleteTodos(todoId: number) {
    return this.http.delete(`${environment.baseUrl}deleteTodos/${todoId}`);
  }

  // Delete Multiple
  deleteMultiple(deleteTodos) {
    return this.http.post(`${environment.baseUrl}deleteMultipleTodos`, deleteTodos);
  }

}

